styx-common/src/main/java/com/spotify/styx/model/ExecutionDescription.java

```diff
@@ -29,15 +29,16 @@
 @AutoMatter
 public interface ExecutionDescription {
 
-  String dockerImage();
-  List<String> dockerArgs();
+  Optional<String> dockerImage();
+  Optional<List<String>> dockerArgs();
```

Comment:
It this distinction something we want to introduce? The difference between an empty list and an empty optional>